pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -472,47 +464,69 @@ func (m *managerImpl) localStorageEviction(pods []*v1.Pod) []*v1.Pod {
 			continue
 		}
 
-		if m.containerOverlayLimitEviction(podStats, pod) {
+		if m.containerScratchLimitEviction(podStats, pod) {
 			evicted = append(evicted, pod)
 		}
 	}
 
 	return evicted
 }
 
+func getPodContainersTotalStorageLimits(pod *v1.Pod) resource.Quantity {
+	containersTotalStorageLimits := resource.Quantity{Format: resource.BinarySI}
+	for _, container := range pod.Spec.Containers {
+		containersTotalStorageLimits.Add(*container.Resources.Limits.StorageScratch())
+	}
+	for _, initContainer := range pod.Spec.InitContainers {
+		if containersTotalStorageLimits.Cmp(*initContainer.Resources.Limits.StorageScratch()) < 0 {
+			containersTotalStorageLimits = *initContainer.Resources.Limits.StorageScratch()
+		}
+	}
+	return containersTotalStorageLimits
+}
+
 func (m *managerImpl) emptyDirLimitEviction(podStats statsapi.PodStats, pod *v1.Pod) bool {
 	podVolumeUsed := make(map[string]*resource.Quantity)
 	for _, volume := range podStats.VolumeStats {
 		podVolumeUsed[volume.Name] = resource.NewQuantity(int64(*volume.UsedBytes), resource.BinarySI)
 	}
+	emptyDirTotalUsed := resource.Quantity{Format: resource.BinarySI}
 	for i := range pod.Spec.Volumes {
 		source := &pod.Spec.Volumes[i].VolumeSource
 		if source.EmptyDir != nil {
 			size := source.EmptyDir.SizeLimit
 			used := podVolumeUsed[pod.Spec.Volumes[i].Name]
+			emptyDirTotalUsed.Add(*used)
 			if used != nil && size.Sign() == 1 && used.Cmp(size) > 0 {
 				// the emptyDir usage exceeds the size limit, evict the pod
 				return m.evictPod(pod, v1.ResourceName("EmptyDir"), fmt.Sprintf("emptyDir usage exceeds the limit %q", size.String()))
 			}
 		}
 	}
+	containersTotalStorageLimits := getPodContainersTotalStorageLimits(pod)
+	if !containersTotalStorageLimits.IsZero() && containersTotalStorageLimits.Cmp(emptyDirTotalUsed) < 0 {
```

Comment:
This pod-level check should include all containers usage and emptyDir usage, not just only total emptyDir usage right?
To make it more clear, maybe have a separate function podLimitEviction()..